pennylane/resource/resource.py

```diff
@@ -167,3 +170,31 @@ def __eq__(self, other: "Resources") -> bool:
                 self.gate_types == other.gate_types,
             ),
         )
+
+
+def count_resources(tape: QuantumTape, shots: int) -> Resources:
+    """Given a quantum workflow (tape) and number of samples, return the resources used.
```

Comment:
This can be more descriptive. Maybe explicitly mention that it counts resources for a quantum tape that contains both standard and custom operations.

Reply:
Added